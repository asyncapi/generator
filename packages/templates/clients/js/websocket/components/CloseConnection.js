import { Text } from '@asyncapi/generator-react-sdk';

export function CloseConnection() {
  return (
    <Text indent={2}>
      {
        `// Method to close the WebSocket connection
close() {
  if (this.websocket) {
    this.websocket.close();
    console.log('WebSocket connection closed.');
  }
}`
      }
    </Text>
  );
}
