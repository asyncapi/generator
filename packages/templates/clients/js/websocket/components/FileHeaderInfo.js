import { Text } from '@asyncapi/generator-react-sdk';

export function FileHeaderInfo({ info, server }) {
  return (
    <Text>
      <Text>
        {'//////////////////////////////////////////////////////////////////////'}
      </Text>
      
      <Text>
        {'//'}
      </Text>

      <Text>
        {'//'} {info.title()} - {info.version()}
      </Text>

      <Text>
        {'//'} Protocol: {server.protocol()}
      </Text>

      <Text>
        {'//'} Host: {server.host()}
      </Text>

      {server.hasPathname() && (
        <Text>
          {'//'} Path: {server.pathname()}
        </Text>
      )}

      <Text>
        {'//'}
      </Text>

      <Text>
        {'//////////////////////////////////////////////////////////////////////'}
      </Text>
    </Text>
  );
}