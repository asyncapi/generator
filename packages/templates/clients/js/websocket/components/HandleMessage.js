import { Text } from '@asyncapi/generator-react-sdk';

export function HandleMessage() {
  return (
    <Text newLines={2} indent={2}>
      {
        `// Method to handle message with callback
handleMessage(message, cb) {
  if (cb) cb(message);
}`
      }
    </Text>
  );
}
