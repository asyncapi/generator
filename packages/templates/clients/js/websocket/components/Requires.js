import { Text } from '@asyncapi/generator-react-sdk';

export function Requires() {
  return (
    <Text>
      <Text>
        const WebSocket = require('ws');
      </Text>
    </Text>
  );
}

