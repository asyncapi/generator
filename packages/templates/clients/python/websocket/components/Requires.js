import { Text } from '@asyncapi/generator-react-sdk';

export function Requires() {
  return (
    <Text>
      <Text>
        {`import json
import certifi
import threading
import websocket`}
      </Text>
    </Text>
  );
}

