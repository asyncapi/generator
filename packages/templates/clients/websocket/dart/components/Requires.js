import { Text } from '@asyncapi/generator-react-sdk';

export function Requires() {
  return (
    <Text>
      <Text>
        {`import 'dart:convert';
import 'package:web_socket_channel/web_socket_channel.dart';`
        }
      </Text>
    </Text>
  );
}

