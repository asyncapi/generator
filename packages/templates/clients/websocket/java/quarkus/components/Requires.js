import { Text } from '@asyncapi/generator-react-sdk';

export function Requires() {
  return (
    <Text>
      <Text>
        {`
package com.asyncapi;

import io.quarkus.websockets.next.WebSocketClient;
import io.quarkus.websockets.next.WebSocketClientConnection;
import io.quarkus.websockets.next.OnOpen;
import io.quarkus.websockets.next.OnClose;
import io.quarkus.websockets.next.OnError;
import io.quarkus.websockets.next.OnTextMessage;
import io.quarkus.websockets.next.CloseReason;
import jakarta.inject.Inject;
import io.quarkus.logging.Log;`}
      </Text>
    </Text>
  );
}

