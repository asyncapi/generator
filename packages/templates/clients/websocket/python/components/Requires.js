import { Text } from '@asyncapi/generator-react-sdk';

export function Requires({ query }) {
  return (
    <Text>
      <Text>
        {`import json
import certifi
import threading
import websocket
${query ? 'import os' : ''}
${query ? 'from urllib.parse import urlencode' : ''}`}
      </Text>
    </Text>
  );
}

