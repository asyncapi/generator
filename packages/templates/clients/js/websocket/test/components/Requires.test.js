import { Requires } from '../../components/Requires';

describe('testing of requires function', () => {
  jest.setTimeout(100000);

  test('render websockts require statement correctly', () => {
    const wrapper = Requires();
    expect(wrapper).toMatchSnapshot();
  });
});
