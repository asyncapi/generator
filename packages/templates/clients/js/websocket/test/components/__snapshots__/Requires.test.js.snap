// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing of requires function render websockts require statement correctly 1`] = `
<Text>
  <Text>
    const WebSocket = require('ws');
  </Text>
</Text>
`;
